pkgs/README.md: classify and add rules to the naming convention (#340452)
NixOS/nixpkgs: pkgs/by-name/README.md

```

> Where `some-package` is the variable name of the package and `so` is the lowercased 2-letter prefix of it.

Author: We just told people there are several terms behind "package name", mainly `pname` and "variable name" (or "attribute name" as you suggested), and they can have potential difference in specific situations (like prefixed by a digit).

> Where `some-package` is the variable name of the package and `so` is the lowercased 2-letter prefix of it.

Reviewer: Right...

```suggestion
Where `some-package` is the attribute name corresponding to the package, and `so` is the lowercase 2-letter prefix of the attribute name.
```
